src/main/java/com/hillel/items_exchange/controller/SubcategoryController.java

```diff
@@ -0,0 +1,75 @@
+package com.hillel.items_exchange.controller;
+
+import com.hillel.items_exchange.dto.SubcategoriesNamesDto;
+import com.hillel.items_exchange.exception.InvalidCategoryVoException;
+import com.hillel.items_exchange.exception.InvalidSubcategoryVoException;
+import com.hillel.items_exchange.service.SubcategoryService;
+import com.hillel.items_exchange.util.StringUtils;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.Collections;
+
+@RestController
+@RequestMapping("/subcategory")
+@CrossOrigin(origins = "http://localhost:4200")
+@RequiredArgsConstructor
+@Slf4j
+public class SubcategoryController {
+
+    private static final String NAME_OF_CLASS = "IN the SubcategoryController: ";
+    private final SubcategoryService subcategoryService;
+
+    @GetMapping("/{category_id}/names")
+    public ResponseEntity<SubcategoriesNamesDto> getSubcategoryNamesByCategoryId(@PathVariable("category_id") Long id) {
+        SubcategoriesNamesDto subcategoriesNames = subcategoryService.findSubcategoryNamesByCategoryId(id);
+        if (subcategoriesNames.getSubcategoriesNamesDto().equals(Collections.emptyList())) {
+            throw new EntityNotFoundException(StringUtils.NO_SUBCATEGORIES_BY_CATEGORY_ID + id);
```

Comment:
Please use exceptions to describe unexpected situation neither showing empty result.